nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -604,21 +607,27 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   // We'll discard points on the plan that are outside the local costmap
   nav2_costmap_2d::Costmap2D * costmap = costmap_ros_->getCostmap();
   const double max_costmap_dim = std::max(costmap->getSizeInCellsX(), costmap->getSizeInCellsY());
-  const double max_transform_dist = max_costmap_dim * costmap->getResolution() / 2.0;
+  const double max_costmap_extent = max_costmap_dim * costmap->getResolution() / 2.0;
```

Comment:
@SteveMacenski we have a choice on whether to use the max costmap extent, which could include points that are actually outside of the costmap, or we could use the min costmap extent, which could exclude points that are actually in the costmap. If we wanted to be exact, we would have to transform the whole path first, and then prune based on the costmap bounding box in the robot frame.

Reply:
Part of the point here is not to transform any thing we don't have to, so I don't think a pre-transform would be a good idea. 

I think this is probably fine as is, I don't think there's a compelling reason to over-complicate this, no? Even if its slightly over, its unlikely to be used as a look ahead point. Even if it is used, the collision checker will check up until the edge of the costmap and if its off the costmap, will return a warning

Reply:
Ok sounds good.

Reply:
I will add a comment explaining that a little more.